tests/virt_autotest/uefi_guest_verification.pm

```diff
@@ -0,0 +1,100 @@
+package uefi_guest_verification;
+
+use base 'virt_feature_test_base';
+use strict;
+use warnings;
+use POSIX 'strftime';
+use File::Basename;
+use testapi;
+use IPC::Run;
+use utils;
+use virt_utils;
+use virt_autotest::common;
+use virt_autotest::utils;
+
+sub run_test {
+    my $self = shift;
+
+    $self->check_guest_bootloader($_) foreach (keys %virt_autotest::common::guests);
+    if (is_kvm_host) {
+        record_soft_failure("In order to implement pm features, current kvm virtual machine uses uefi firmware that does not support PXE/HTTP boot and secureboot", "bsc#1182886 UEFI virtual machine boots with trouble");
+        #$self->check_guest_uefi_boot($_) foreach (keys %virt_autotest::common::guests);
+        #$self->check_guest_secure_boot($_) foreach (keys %virt_autotest::common::guests);
+    }
+    else {
+        record_soft_failure("UEFI implementation for xen fullvirt uefi virtual machine is incomplete", "bsc#1184936 Xen fullvirt lacks of complete support for UEFI");
+    }
+    $self->check_guest_pmsuspend_enabled;
+    return $self;
+}
+
+sub check_guest_bootloader {
+    my ($self, $guest_name) = @_;
+
+    my $ssh_command_prefix = "ssh -vvv -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no";
+    script_retry("$ssh_command_prefix root\@$guest_name efibootmgr -v");
+    script_retry("$ssh_command_prefix root\@$guest_name mokutil --sb-state");
+    return $self;
+}
+
+sub check_guest_uefi_boot {
+    my ($self, $guest_name) = @_;
+
+    my $ssh_command_prefix = "timeout 30 ssh -vvv -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no";
+    assert_script_run("$ssh_command_prefix root\@$guest_name efibootmgr -v | grep -iE \"uefi.*pxev6\"");
+    assert_script_run("$ssh_command_prefix root\@$guest_name efibootmgr -v | grep -iE \"uefi.*HTTPv6\"");
```

Comment:
What about adding checks for the installed sles guest os , whether they show up correctly in bootorder and currentboot items of the "efibootmgr -v" output? IMHO, uefi guest users care about this a lot, especially for multi-boot systems which installs several os on the guest.

Reply:
I added new checking:
check_guest_bootcurrent